packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -123,11 +123,16 @@ library LibProving {
         fc.provenAt = uint64(block.timestamp);
         fc.gasUsed = evidence.gasUsed;
 
-        IProofVerifier(resolver.resolve("proof_verifier", false)).verifyProofs({
-            blockId: blockId,
-            blockProofs: evidence.proofs,
-            instance: getInstance(config, resolver, evidence)
-        });
+        bytes32 instance = getInstance(
+            evidence,
+            resolver.resolve("signal_service", false),
```

Comment:
@Brechtpd We may also skip hashing these addresses as they will never change (can be upgraded). What do you think?

Reply:
Yeah I'm okay with skipping and hard-coding it in the circuits. Hashing these values here when the proof is being submitted was actually flawed because it made the block non-deterministic if the value changed, so even if we would want to hash these values, it should be done at the proposer side so the proof always remains the same even if the config changes.